drivers/ble/sphero/sphero_r2d2_driver.go

```diff
@@ -0,0 +1,396 @@
+}
+
+// SetRawMotorValues allows you to take over one or both of the motor output values, instead of having the stabilization
+// system control them. Each motor (left and right) requires a mode and a power value from 0-255.
+// MotorModes Brake and Ignore are not supported on the R2.
+func (d *R2D2Driver) SetRawMotorValues(lmode MotorModes, lpower uint8, rmode MotorModes, rpower uint8) {
+	// did: 22, cid: 1
+	d.sendCraftPacket([]uint8{uint8(lmode), lpower, uint8(rmode), rpower}, 0x16, 0x01)
+}
+
+// SetBackRGB sets the back R2D2 dome LED to the given r, g, and b values
+func (d *R2D2Driver) SetBackRGB(r uint8, g uint8, b uint8) {
+	// did: 26, cid: 14
+	//	def set_leds(self, mapping: Dict[IntEnum, int]):
+	//	mask = 0
+	//	led_values = []
+	//	for e in self.__toy.LEDs:
+	//		if e in mapping:
+	//			mask |= 1 << e
+	//			led_values.append(mapping[e])
+	//	self.__toy.set_all_leds_with_16_bit_mask(mask, led_values)
+	// IO._encode(toy, 14, proc, [*to_bytes(mask, 2), *values])
+	d.sendCraftPacket([]uint8{0, 119, r, g, b, r, g, b}, 0x1A, 0x0E)
+}
+
+func (d *R2D2Driver) PerformLegAction(action LegAction) {
+	// did: 23, cid: 13
+	d.sendCraftPacket([]uint8{uint8(action)}, 0x17, 0x0D)
+}
+
+// SetDomePosition pos can be -160 to 180
+func (d *R2D2Driver) SetDomePosition(pos float32) {
+	// did: 23, cid: 15
+	d.sendCraftPacket(spherocommon.Float32ToBytes(pos), 0x17, 0x0F)
+}
+
+// PlaySound where playback is PlaybackImmediate, PlaybackIfNotPlaying or PlaybackAfterCurrentSound
+func (d *R2D2Driver) PlaySound(sound Audio, playback Playback) {
+	// did: 26, cid: 7
+	d.sendCraftPacket(append(spherocommon.Uint16ToBytes(uint16(sound)), uint8(playback)), 0x1A, 0x07)
+}
+
+func (d *R2D2Driver) PlayAnimation(anima Animation) {
+	// did: 23, cid: 5
+	d.sendCraftPacket(spherocommon.Uint16ToBytes(uint16(anima)), 0x17, 0x05)
+}
+
+// Stop tells the R2D2 to stop
+func (d *R2D2Driver) Stop() {
+	d.Roll(0, 0)
+}
+
+// Sleep says Go to sleep
+func (d *R2D2Driver) Sleep() {
+	// did: 19, cid: 1
+	d.sendCraftPacket([]uint8{}, 0x13, 0x1)
+}
+
+// SetDataStreamingConfig passes the config to the sphero to stream sensor data
+// TODO get this working for R2
+func (d *R2D2Driver) SetDataStreamingConfig(dsc spherocommon.DataStreamingConfig) error {
+	buf := new(bytes.Buffer)
+	if err := binary.Write(buf, binary.BigEndian, dsc); err != nil {
+		return err
+	}
+	d.sendCraftPacket(buf.Bytes(), 0x02, 0x11)
+	return nil
+}
+
+// initialize tells driver to get ready to do work
+func (d *R2D2Driver) initialize() error {
+	if err := d.antiDOSOff(); err != nil {
+		return err
+	}
+	if err := d.Wake(); err != nil {
+		return err
+	}
+	// subscribe to Sphero response notifications
+	if err := d.Adaptor().Subscribe(r2ResponseChara, d.handleResponses); err != nil {
+		return err
+	}
+
+	go func() {
+		for {
+			packet := <-d.packetChannel
+			err := d.writeCommand(packet)
+			if err != nil {
+				d.Publish(d.Event(spherocommon.ErrorEvent), err)
+			}
```

Comment:
```suggestion
			if err := d.writeCommand(packet); err != nil {
				d.Publish(d.Event(spherocommon.ErrorEvent), err)
			}
```
prevents misuse of "err" later